lib/update/cluster/automatic.go

```diff
@@ -43,6 +43,21 @@ func AutomaticUpgrade(ctx context.Context, localEnv, updateEnv *localenv.LocalEn
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	siteOp := (*ops.SiteOperation)(operation)
+
+	plan, err := fsm.GetOperationPlan(updateEnv.Backend, siteOp.Key())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Check if the upgrade is already in progress, which would indicate an
+	// agent is already running. Return without creating a new agent.
+	for _, phase := range plan.Phases {
+		if phase.IsInProgress() {
```

Comment:
I think we should be checking for "not unstarted" phases. Otherwise, this will not catch scenarios where a phase is "failed" or "completed". So we should basically look for phases in any state other than the initial "unstarted" one.